docs/content/docs/core-concept/transform.md

```diff
@@ -28,5 +28,245 @@ under the License.
 
+# Parameters
+To describe a transform, the follows are required:
+
+| Parameter    | Meaning                                            | Optional/Required |
+|--------------|----------------------------------------------------|-------------------|
+| source-table | Source table id, supports regular expressions      | required          |
+| projection   | Projection rule, supports syntax similar to the select clause in SQL        | optional          |
+| filter       | Filter rule, supports syntax similar to the where clause in SQL        | optional          |
+| primary-keys | Sink table primary keys, it is separated by English commas        | optional          |
+| partition-keys | Sink table partition keys, it is separated by English commas        | optional          |
+| table-options | Sink table options, when automatically creating a table, it is added to the configuration of the table creation statement  | optional          |
+| description  | Transform rule description(a default value provided) | optional          |
+
+A transform module can contain a list of source-table rules.
+
+# Metadata Fields
+## Fields definition
+There are three hidden columns used to describe metadata information. They only take effect when explicitly referenced in the transform rules.
+
+| Field               | Data Type | Description                                                     |
+|--------------------|-----------|--------------------------------------------------------------|
+| __namespace_name__ | String    | Name of the namespace that contain the row.                  |
+| __schema_name__    | String    | Name of the schema that contain the row.                     |
+| __table_name__     | String    | Name of the table that contain the row.                      |
+
+## Metadata relationship
+
+| Type                 | Namespace | SchemaName | Table |
+|----------------------|-----------|------------|-------|
+| JDBC                 | Catalog   | Schema     | Table |
+| Debezium             | Catalog   | Schema     | Table |
+| MySQL                | Database  | -          | Table |
+| Postgres             | Database  | Schema     | Table |
+| Oracle               | -         | Schema     | Table |
+| Microsoft SQL Server | Database  | Schema     | Table |
+| StarRocks            | Database  | -          | Table |
+| Doris                | Database  | -          | Table |
+
+# Functions
+Using Calcite to implement function parsing, and using Janino script to implement function calls.
+
+## Comparison Functions
+
+| Function             | Janino Code                 | Description                                                     |
+|----------------------|-----------------------------|-----------------------------------------------------------------|
+| value1 = value2      | valueEquals(value1, value2) | Returns TRUE if value1 is equal to value2; returns FALSE if value1 or value2 is NULL. |
```

Comment:
It would be nice to put code pieces in backslashes like `value1 = value2`, but not necessarily required. Ignore this comment if it takes too much effort.